tests/roles/dataplane_adoption/tasks/main.yaml

```diff
@@ -75,7 +75,7 @@
           ssh \
             -i {{ edpm_privatekey_path | default("/tmp/ansible_private_key") }} \
             {{ edpm_user }}@"${computes[$name]}" \
-            "echo $uuid > /var/lib/nova/compute_id"
+            "echo $uuid | sudo tee /var/lib/nova/compute_id"
```

Comment:
if this keeps read access for the nova-compute service to this file then I'm fine with it.

Reply:
compute_id is registered with this command. But the id does not match with the one registered in compute service list and computes state is down.
https://privatebin.corp.redhat.com/?2a67ecedadf5a3a2#3qmFnemnvWxERgykPRhiE3FQ3JjR96hiTRzftm49ohSh

Reply:
@bogdando @gibizer any idea why the compute state is down?

Reply:
@archana203 could you point me to the job logs?

Reply:
Without logs or troubleshooting I can only guess.
A) the nova-compute service cannot read the compute_id file as it is now written via sudo
B) for some reason the openstack client is different in this case and instead of defaulting to the maximum nova api microversion it defaults to the minimum. If this is the case then you can add `--os-compute-api-version 2.53` to the openstack hypervisor command to force the newer microversion. (Before 2.53 the hypervisor show CLI returned the 'id' not the 'uuid' of the compute node object)

Reply:
I looked at the deployment provided by @archana203 . The OSP18 deployment failed at network config
```
TASK [osp.edpm.edpm_users : Ensure user is present on the host [ openvswitch ]] ***
Monday 22 January 2024  18:18:17 +0000 (0:00:00.630)       0:00:03.397 ********
fatal: [compute-0]: FAILED! => {"changed": false, "msg": "usermod: user openvswitch is currently used by process 15864\n", "name": "openvswitch", "rc": 8}
fatal: [compute-1]: FAILED! => {"changed": false, "msg": "usermod: user openvswitch is currently used by process 15950\n", "name": "openvswitch", "rc": 8}
```
So the new nova_compute container is not yet deployed.

The compute_id is generate properly. It has o+r rights so nova-compute will be able to read it.

Reply:
there was a fix https://github.com/openstack-k8s-operators/edpm-ansible/pull/540

Reply:
does it mean sudo isn't needed anymore?

Reply:
no sudo is still required

but the issue gibi noted is fixed.

it was unrelated to this change

Reply:
/var/lib/nova/compute_id is chown'd by kolla when the nova_compute container starts so this will become owned by nova at that point so yes nova will have read access to it.